### PR TITLE
Fix functional component props

### PR DESCRIPTION
**Pull request by mhaas**
Another one. See:

https://github.com/brigand/babel-plugin-flow-react-proptypes/pull/88/commits/f2b1e09f8f2c97bb68c9f5a1b554a876a5c85109

This MR shows all commits of #87 because I pushed before you merged that one, but it really is only the one above + a merge commit.

### CODE REVIEW

**Comment by brigand** [comment]
Could you rebase this on master?

**Comment by brigand** [comment]
It looks like this is setting C.propTypes to a function; it needs to be an object.

File: src/__tests__/__snapshots__/object-props-test.js.snap [hunk 1]
```diff
@@ -3,5 +3,6 @@ exports[`test object-props 1`] = `
 
 var Component = function Component(props) {
   return React.createElement(\"div\", null);
-};"
+};
+Component.propTypes = require(\"prop-types\").object;"
```
Comment (mhaas) [inline src/__tests__/__snapshots__/object-props-test.js.snap]: @brigand I agree about C.proptypes being an object; however, is this usage here OK?

**Comment by mhaas** [comment]
Rebased.

File: src/__tests__/__snapshots__/object-props-test.js.snap (hunk 1 above)
Comment (brigand) [inline src/__tests__/__snapshots__/object-props-test.js.snap]: Nope, gives:

```
Failed prop type: The prop `isRequired` is marked as required in `App`, but its value is `undefined`.
```

It needs to be an object.

Comment (mhaas) [inline src/__tests__/__snapshots__/object-props-test.js.snap]: What would be the best way to translate this? If I do Component.propTypes = {}, will that be useful?

Comment (brigand) [inline src/__tests__/__snapshots__/object-props-test.js.snap]: I'd just omit it.

**Comment by mhaas** [comment]
@brigand Should be good now. I am omitting the non-useful cases now.

**Comment by brigand** [comment]
Thanks!

**Comment by mhaas** [comment]
You're welcome.

Funnily enough, this actually adds support for setting Foo.propTypes directly from an import, i.e. makes the case in `top-level-import.js` work. I just spent quite some time wondering how this ever worked or rather, why it was only broken for functional components. Turns out: top-level imports never worked, I was just too focused on intersections and nested usage of imported types to notice.

**Pull request merged**